lib/settings-formats.nix

```diff
@@ -0,0 +1,51 @@
+{ lib }: rec {
+  /*
+  Every entry in this file represents a format for configuration files of
+  programs, used for `settings` options. Each entry has to look as follows:
+
+    <format> = {
+
+      # The module system type most suitable for representing such a format
+      type = ...;
+
+      # A function taking an instance of such a type and turning it into a string
+      generate = ...;
+
+    }
+  */
+
+  json = {
+    type = with lib.types; let
+      value = nullOr (oneOf [
+        bool
+        int
+        float
+        str
+        (attrsOf value)
+        (listOf value)
+      ]) // {
```

Comment:
This is not recursive, is it?

Reply:
It is set on `value`, which is used in the definition itself, but the `description` doesn't refer to the subtypes, so it won't be infinitely long

Reply:
Ah, missed that.